### PR TITLE
Register vaapih265dec H265 decoder

### PR DESCRIPTION
**Pull request by rgonzalezfluendo**

### CODE REVIEW

File: decoders/gstreamer.py [hunk 1]
```diff
@@ -70,3 +70,13 @@ class FluendoH265Gst010Decoder(GStreamer010):
     description = "Fluendo H.265 software decoder for GStreamer 0.10"
     codec = Codec.H265
     decoder_element = ' h265parse ! fluh265dec '
+
+
+@register_decoder
+class FluendoH265Gst10Decoder(GStreamer10):
```
Comment (ylatuya) [inline decoders/gstreamer.py:76]: The name of this class is duplicated. The CI correctly complained about it with `decoders/gstreamer.py:76:0: E0102: class already defined line 58 (function-redefined)` :)

https://dev.azure.com/fluendo/Fluxion/_build/results?buildId=13&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=5caf77c8-9b10-50ef-b5c7-ca89c63e1c86

File: decoders/gstreamer.py [hunk 2]
```diff
@@ -70,3 +70,13 @@ class FluendoH265Gst010Decoder(GStreamer010):
...
+    '''Community H.265 VAAPI decoder implementation for GStreamer 1.0'''
+    name = 'GStreamer-H265-VAAPI-Gst10'
+    description = "Community H.265 VAAPI decoder implementation for GStreamer 1.0"
+    codec = Codec.H265
+    decoder_element = ' h265parse ! vaapih265dec '
+    caps = 'video/x-raw,format=I420'
```
Comment (ylatuya) [inline decoders/gstreamer.py:82]: This will force a colorspace conversion for profiles using 4:4:4

File: decoders/gstreamer.py (hunk 1 above)
Comment (rgonzalezfluendo) [reply decoders/gstreamer.py]: I was testing the CI :wink:

File: decoders/gstreamer.py (hunk 2 above)
Comment (rgonzalezfluendo) [reply decoders/gstreamer.py:82]: Default format of `vaapih265dec` f is `NV12`:

```
gst-launch-1.0 -v filesrc location=resources/JCT-VC-HEVC_V1/WPP_F_ericsson_MAIN_2/WPP_F_ericsson_MAIN_2.bit ! h265parse ! vaapih265dec ! video/x-raw ! filesink location=/dev/null
Setting pipeline to PAUSED ...
Pipeline is PREROLLING ...
Got context from element 'vaapidecode_h265-0': gst.gl.GLDisplay=context, gst.gl.GLDisplay=(GstGLDisplay)"\(GstGLDisplayX11\)\ gldisplayx11-0";
Got context from element 'vaapidecode_h265-0': gst.vaapi.Display=context, gst.vaapi.Display=(GstVaapiDisplay)"\(GstVaapiDisplayGLX\)\ vaapidisplayglx0";
/GstPipeline:pipeline0/GstH265Parse:h265parse0.GstPad:src: caps = video/x-h265, width=(int)192, height=(int)240, chroma-format=(string)4:2:0, bit-depth-luma=(uint)8, bit-depth-chroma=(uint)8, parsed=(boolean)true, stream-format=(string)hvc1, alignment=(string)au, profile=(string)main, tier=(string)main, level=(string)2, codec_data=(buffer)0101400000000000000000003cf000fcfdf8f800000f03200001001940010c01ffff0140000003000003000003000003003c94902421000100324201010140000003000003000003000003003ca018203c7f96526491b618588aa93130cbecfaf37e59f5e1446272ef0003e222000100074401e1a564c18d
/GstPipeline:pipeline0/GstVaapiDecode_h265:vaapidecode_h265-0.GstPad:sink: caps = video/x-h265, width=(int)192, height=(int)240, chroma-format=(string)4:2:0, bit-depth-luma=(uint)8, bit-depth-chroma=(uint)8, parsed=(boolean)true, stream-format=(string)hvc1, alignment=(string)au, profile=(string)main, tier=(string)main, level=(string)2, codec_data=(buffer)0101400000000000000000003cf000fcfdf8f800000f03200001001940010c01ffff0140000003000003000003000003003c94902421000100324201010140000003000003000003000003003ca018203c7f96526491b618588aa93130cbecfaf37e59f5e1446272ef0003e222000100074401e1a564c18d
Redistribute latency...
/GstPipeline:pipeline0/GstVaapiDecode_h265:vaapidecode_h265-0.GstPad:src: caps = video/x-raw, format=(string)NV12, width=(int)192, height=(int)240, interlace-mode=(string)progressive, multiview-mode=(string)mono, multiview-flags=(GstVideoMultiviewFlagsSet)0:ffffffff:/right-view-first/left-flipped/left-flopped/right-flipped/right-flopped/half-aspect/mixed-mono, pixel-aspect-ratio=(fraction)1/1, chroma-site=(string)jpeg, colorimetry=(string)bt601, framerate=(fraction)0/1
/GstPipeline:pipeline0/GstCapsFilter:capsfilter0.GstPad:src: caps = video/x-raw, format=(string)NV12, width=(int)192, height=(int)240, interlace-mode=(string)progressive, multiview-mode=(string)mono, multiview-flags=(GstVideoMultiviewFlagsSet)0:ffffffff:/right-view-first/left-flipped/left-flopped/right-flipped/right-flopped/half-aspect/mixed-mono, pixel-aspect-ratio=(fraction)1/1, chroma-site=(string)jpeg, colorimetry=(string)bt601, framerate=(fraction)0/1
/GstPipeline:pipeline0/GstFileSink:filesink0.GstPad:sink: caps = video/x-raw, format=(string)NV12, width=(int)192, height=(int)240, interlace-mode=(string)progressive, multiview-mode=(string)mono, multiview-flags=(GstVideoMultiviewFlagsSet)0:ffffffff:/right-view-first/left-flipped/left-flopped/right-flipped/right-flopped/half-aspect/mixed-mono, pixel-aspect-ratio=(fraction)1/1, chroma-site=(string)jpeg, colorimetry=(string)bt601, framerate=(fraction)0/1
/GstPipeline:pipeline0/GstCapsFilter:capsfilter0.GstPad:sink: caps = video/x-raw, format=(string)NV12, width=(int)192, height=(int)240, interlace-mode=(string)progressive, multiview-mode=(string)mono, multiview-flags=(GstVideoMultiviewFlagsSet)0:ffffffff:/right-view-first/left-flipped/left-flopped/right-flipped/right-flopped/half-aspect/mixed-mono, pixel-aspect-ratio=(fraction)1/1, chroma-site=(string)jpeg, colorimetry=(string)bt601, framerate=(fraction)0/1
Pipeline is PREROLLED ...
Setting pipeline to PLAYING ...
New clock: GstSystemClock
Got EOS from element "pipeline0".
Execution ended after 0:00:00.019906169
Setting pipeline to NULL ...
Freeing pipeline ...

```

Comment (ylatuya) [reply decoders/gstreamer.py:82]: I see, if we want to support all the profiles, we will need to set the caps filter based on the chroma-format field from the h265parser caps.

**Comment by ylatuya** [comment]
In master the GStreamer base class changed trying to unify the namings (https://github.com/fluendo/fluxion/commit/1f1c19b76a73e00eaf7d25d67144026b88e6b550).
Can you please change this implementation to follow the new way?
This one should have:
- provider: GStreamer
- api: VA-API
- gst_api: 1.0

**Comment by pamarcos** [comment]
Closing in favor of https://github.com/fluendo/fluxion/pull/14

**Pull request closed without merging**